src/Magento/FunctionalTestingFramework/Allure/Adapter/MagentoAllureAdapter.php

```diff
@@ -309,9 +315,28 @@ function () use ($rootStep, $formattedSteps) {
 
         $this->getLifecycle()->getStepStorage()->put($rootStep);
 
+        $this->addAttachmentEvent($testEvent);
```

Comment:
What changed in codeception that requires this new line + functions?

Reply:
testEnd of codeception has changed to this, it takes a TestEvent parameter. I see any changes due to this addition in test runs. I'm not 100% sure this change does, any thoughts? TestEvent needed to be used in overriding function in MagentoAllureAdapter, hence the addition.
```
    public function testEnd(TestEvent $testEvent)
    {
        // attachments supported since Codeception 3.0
        if (version_compare(Codecept::VERSION, '3.0.0') > -1 && $testEvent->getTest() instanceof Cest) {
            $artifacts = $testEvent->getTest()->getMetadata()->getReports();
            foreach ($artifacts as $name => $artifact) {
                Allure::lifecycle()->fire(new AddAttachmentEvent($artifact, $name, null));
            }
        }
        $this->getLifecycle()->fire(new TestCaseFinishedEvent());
    }
```